hdf5.spec

```diff
@@ -300,10 +300,15 @@ do
   mkdir $mpi
   pushd $mpi
 %if (0%{?suse_version} >= 1500)
-  module load gnu-$mpi
+  if [ "$mpi" == "openmpi3" ]; then
+    module load gnu-openmpi
+  else
+    module load gnu-$mpi
+  fi
```

Comment:
Yeah, seeing this all come back, I think the SUSE/RH conditionals should be around the %{mpi_list} construction, not it's evaluation.

Or alternatively (and better yet), put all of this into a macro called `load_mpi` or something so that the execution locations, like this one are much cleaner with just: `load_mpi $mpi` and let the macro do all the work of figuring out the command and correct name to load, etc.

Reply:
Phil had recommended the same macro but I could not get it to pass.  The changes we made were:

if (0%{?suse_version} >= 1500)	
%global load_module() (if [ "%{1}" == "openmpi3" ]; then module load gnu-openmpi; else module load gnu-%{1}; fi)	
%else	
%global load_module() (module load mpi/%{1}-%{_arch})	
%endif

and it would not build with mpich; so I reverted the changes and it seemed to pass; however the DAOS job with hdf5_mpich is now failing, when it passed previously.

Reply:
So let's resurrect the macro and figure out why it fails.

Reply:
wrapping the macro body in `()` was causing the `module load` to be run in a sub-shell, preventing the environment from being set correctly.